Documentation/api_v3.md

```diff
@@ -1,6 +1,33 @@
 # etcd3 API
 
-TODO: API doc
+TODO: finish API doc
+
+## Key-Value API
+
+Key-Value API is used to manipulate key-value pairs stored inside etcd. The key-value API is defined as a gRPC service[kv-service]. The Key-Value pair is defined as a structured data in [protobuf format][kv-proto].
+
+### Key-Value Pair
+
+Key-Value pair is the smallest unit that Key-Value API can manipulate. Each key-value pair has a few fields listed below.
+
+```protobuf
+message KeyValue {
+  bytes key = 1;
+  int64 create_revision = 2;
+  int64 mod_revision = 3;
+  int64 version = 4;
+  bytes value = 5;
+  int64 lease = 6;
+}
+```
+
+* Key - key in bytes. An empty key is not allowed.
+* Value - value in bytes.
+* Version - version is the version of the key. A deletion resets the version to zero and any modification of the key increases its version.
+* Create_Revision - revision of the last creation on the key.
```

Comment:
Revision is the update sequence number of the kv store. Version is the update sequence number of the key since being created.

Ex:
```sh
ETCDCTL_API=3 ./etcdctl put testkey 123
# OK
ETCDCTL_API=3 ./etcdctl -w json get testkey
# {"key":"dGVzdGtleQ==","create_revision":6,"mod_revision":6,"version":1,"value":"MTIz"}
ETCDCTL_API=3 ./etcdctl put testkey 456
# OK
ETCDCTL_API=3 ./etcdctl -w json get testkey    
# {"key":"dGVzdGtleQ==","create_revision":6,"mod_revision":7,"version":2,"value":"NDU2"}
```